bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/cli/CliProvider.java

```diff
@@ -285,6 +285,16 @@ public static Object getFiniteBalValue(CliArg arg, BFiniteType finiteType, Varia
         }
     }
 
+    @Override
+    public Optional<ConfigValue> getAsTupleAndMark(Module module, VariableKey key) {
+        CliArg cliArg = getCliArg(module, key);
+        if (cliArg.value == null) {
+            return Optional.empty();
```

Comment:
Can we directly throw the error?

Reply:
We can't throw a direct error here. Because, if the user uses both CLI and `Config.toml` to configure, and does not provide unsupported type via CLI arguments, it should return empty without throwing an error. 
I added a few unit tests that cover this use case.